evo/tools/tf_cache.py

```diff
@@ -108,50 +108,39 @@ def from_bag(self, reader: Rosbag1Reader, topic: str = "/tf",
             self.topics.append(tf_topic)
         self.bags.append(reader.path.name)
 
-    def lookup_trajectory(
-        self, parent_frame: str, child_frame: str, start_time: rospy.Time,
-        end_time: rospy.Time,
-        lookup_frequency: float = SETTINGS.tf_cache_lookup_frequency
-    ) -> PoseTrajectory3D:
+    def lookup_trajectory(self, parent_frame: str, child_frame: str,
+                          timestamps: List[rospy.Time]) -> PoseTrajectory3D:
         """
         Look up the trajectory of a transform chain from the cache's TF buffer.
         :param parent_frame, child_frame: TF transform frame IDs
-        :param start_time, end_time: expected start and end time of the
-                                     trajectory in the buffer
+        :param timestamps: timestamps at which to lookup the trajectory poses.
         :param lookup_frequency: frequency of TF lookups between start and end
                                  time, in Hz.
         """
         stamps, xyz, quat = [], [], []
-        step = rospy.Duration.from_sec(1. / lookup_frequency)
-        # Static TF have zero timestamp in the buffer, which will be lower
-        # than the bag start time. Looking up a static TF is a valid request,
-        # so this should be possible.
```

Comment:
Please don't remove existing behavior.
Loading a static transform doesn't work anymore with your commit.

Reply:
sorry for this!
Can you help me in understanding how it was working before?
In which scenario are we loading 0 as end time of a bagfile and a start time bigger than 0?
Isn't it always that the start time is <= that the end time?

Here I was assuming that if someone is giving a timestamp list = [rospy.Time(0)] ...the wanted behavior was still possible.